config_src/infra/FMS1/MOM_diag_manager_infra.F90

```diff
@@ -276,15 +294,15 @@ end function send_data_infra_1d
 logical function send_data_infra_2d(diag_field_id, field, is_in, ie_in, js_in, je_in, &
                                     time, mask, rmask, weight, err_msg)
   integer,                           intent(in) :: diag_field_id !< The diagnostic manager identifier for this field
-  real, dimension(:,:),              intent(in) :: field !< A 2-d array of values being recorded
+  class(*),dimension(:,:),           intent(in) :: field !< A 2-d array of values being recorded
```

Comment:
Why is this needed? And if it is required, why is it not also needed in the caller of this routine?